docker-bake.hcl

```diff
@@ -16,12 +20,26 @@ function "minor" {
     params = [version]
     result = join(".", slice(split(".", version),0,2))
 }
-# result = split(version, ".")[0] + "." + split(version, ".")[1] "." + split(version, ".")[2]
+
 function "patch" {
     params = [version]
     result = join(".", slice(split(".", version),0,3))
 }
 
+function "tag" {
+    params = [tag]
+    result = ["${REPO}:${tag}"]
+}
+
+function "vtag" {
+    params = [semver, variant]
+    result = concat(
+        tag("${major(semver)}${variant}-${formatdate("YYYYMMDDHHMM", timestamp())}"),
```

Comment:
```shell
docker buildx bake -f docker-bake.hcl
[+] Building 0.0s (0/0)
docker-bake.hcl:36
--------------------
  35 |         params = [semver, variant]
  36 | >>>     result = concat(
  37 | >>>         tag("${major(semver)}${variant}-${formatdate("YYYYMMDDHHMM", timestamp())}"),
  38 | >>>         tag("${minor(semver)}${variant}-${formatdate("YYYYMMDDHHMM", timestamp())}"),
  39 | >>>         tag("${patch(semver)}${variant}-${formatdate("YYYYMMDDHHMM", timestamp())}")
  40 | >>>     )
  41 |     }
--------------------
error: docker-bake.hcl:36,14-40,6: Invalid expression; undefined function timestamp
```

Reply:
You need to have a modern docker buildx version. Like 0.9.1 and upwards.

Reply:
I had missed the last update to Docker Desktop. I think you should add the requirement to the Readme.

Reply:
Added.